### PR TITLE
add support to environment variable override

### PR DESCRIPTION
**Pull request by omarcinp**
I know nobody asked me to do this, but my mom always told me "te mandas solo verdad?

So I decided to make this.

when a file named `environment-variables.yaml` is added, the cfg package will try to read value of any key inside that file as an environment variable name and if exists cfg will override the value

### CODE REVIEW

**Review by rascencioc — COMMENTED** [review]
LGTM

File: cfg.go [hunk 1]
```diff
@@ -61,9 +64,77 @@ func (c *configWrapper) mergeEnv(envName string) error {
...
+
+func isFloat(data string) bool {
+	if _, err := strconv.ParseFloat(data, 64); err == nil {
+		return true
+	}
+	return false
+}
+
+func isBool(data string) bool {
+	return (data == "true" || data == "false")
```
Comment (dtoledom) [inline cfg.go]: If you are already using strconv, better use [ParseBool](https://golang.org/pkg/strconv/#ParseBool)

File: cfg.go [hunk 2]
```diff
@@ -61,9 +64,77 @@ func (c *configWrapper) mergeEnv(envName string) error {
...
+}
+
 // Get returns a map[string]interface{} of a given key
-func (c *configWrapper) Get(key string) map[string]interface{} {
-	return c.v.GetStringMap(key)
+func (c *configWrapper) Get(key string) interface{} {
+	d := c.v.AllSettings()
+	v := interface{}(d)
+	path := strings.Split(key, ".")
+	// fmt.Println(path)
```
Comment (dtoledom) [inline cfg.go]: Delete commented code here and below

File: cfg.go [hunk 3]
```diff
@@ -16,6 +18,7 @@ const (
 
 	defaultConfigEnv = "default"
 	localConfigEnv   = "local"
+	envVarConfigFile = "environment-variables"
```
Comment (grevych) [inline cfg.go:21]: Maybe `custom-environment-variables`? -> https://github.com/lorenwest/node-config/wiki/Environment-Variables#custom-environment-variables

File: cfg.go [hunk 4]
```diff
@@ -61,9 +64,77 @@ func (c *configWrapper) mergeEnv(envName string) error {
...
+			}
+		}
+	}
+	return
+}
+
 // Get returns a map[string]interface{} of a given key
-func (c *configWrapper) Get(key string) map[string]interface{} {
-	return c.v.GetStringMap(key)
+func (c *configWrapper) Get(key string) interface{} {
```
Comment (dtoledom) [inline cfg.go:99]: There is some edge cases where I think this implementation does not work.  The following testcase fails: 

```"TestNonExistentConfig": {
			key:               "services.foo.string.foo",
			expectedConfigMap: nil,
		},
```

But don't reinvent the wheel, this logic is already implemented in the [Viper](https://godoc.org/github.com/spf13/viper) library. Simply do `return c.v.Get(key)`

**Review by dtoledom — CHANGES_REQUESTED** [review]
Nice! 💯 thanks for taking the time to add this feature! I left some comments

File: cfg.go [hunk 5]
```diff
@@ -61,9 +64,77 @@ func (c *configWrapper) mergeEnv(envName string) error {
...
+	tempV.SetConfigType(c.params.FileType)
+	tempV.SetConfigName(envVarConfigFile)
+	tempV.AddConfigPath(c.params.Path)
+	if err := tempV.ReadInConfig(); err != nil {
+		return
+	}
+	keys := tempV.AllKeys()
+	for _, key := range keys {
+		envVarName := tempV.GetString(key)
+		if value := os.Getenv(envVarName); value != "" {
```
Comment (mmiranda96) [inline cfg.go]: We should use `os.LookupEnv` instead: in that way we can determine whether the variable doesn't exist or the value is "".

File: cfg.go (hunk 4 above)
Comment (omarcinp) [reply cfg.go:99]: c.v.Get does not returns the full map after overriding it with `Set()` 😢

File: cfg.go (hunk 3 above)
Comment (omarcinp) [reply cfg.go:21]: I think an smaller name is better 🙊

File: cfg_test.go [hunk 6]
```diff
@@ -161,3 +162,57 @@ func TestGet(t *testing.T) {
...
+			key: "services.foo",
+			expectedConfigMap: map[string]interface{}{
+				"string": "foo",
+				"int":    42,
+				"float":  10.5,
+				"bool":   false,
+			},
+		},
+		"TestSingleMissingDeepKey": {
+			key:               "services.foo.string.foo",
```
Comment (dtoledom) [inline cfg_test.go:197]: 👍

**Review by dtoledom — APPROVED** [review]

**Pull request merged**